src/main/java/org/jboss/remoting3/EndpointImpl.java

```diff
@@ -513,12 +513,17 @@ public void handleFailed(final IOException exception, final FutureResult<Connect
             }
 
             public void handleDone(final Connection connection, final FutureResult<ConnectionPeerIdentity> attachment) {
-                try {
-                    final ConnectionPeerIdentity identity = connection.getPeerIdentityContext().authenticate(authenticationConfiguration);
-                    futureResult.setResult(identity);
-                } catch (AuthenticationException e) {
-                    futureResult.setException(e);
-                }
+                worker.execute(() -> {
```

Comment:
@dmlloyd use lambda even with the performance cost, or just create an anonymous Runnable?

Reply:
I think either way is OK.